ParcialProgramacion4/src/Empleado.java

```diff
@@ -37,4 +37,41 @@ public Puesto getPuestoActual() {
         //siempre se cumple que el ultimo agregado es el actual, entonces saco el ultimo
         return this.historialDeCargos.get(historialDeCargos.size() - 1).getPuesto();
     }
+    
+  //SI MAL NO RECUERDO ESTO RECIBE LA CLASE Y UN ENTERO QUE LO EMPAQUETA EN UN INTEGER/.
+    public void agregarHabilidad(Habilidad habilidadExistente,int unaCantidadDeTiempo)
+    {
+        if(habilidades.containsKey(habilidadExistente))  // SI CONTIENE LA POS
+			unaCantidadDeTiempo+=habilidades.get(habilidadExistente); // AL ENC LE PONGO LOS ANIOS???
+		habilidades.put(habilidadExistente,unaCantidadDeTiempo);
+    }
+
+
+    //METODO PARA ELIMINAR HABILIDAD EXISTENTE DE LA HASH DEL EMPLEADO
+    public void eliminarHabilidad(Habilidad habilidadExistente)
+    {
+		Enumeration<Habilidad> enumH=habilidades.keys(); // me devolvia todo esto
+		while(enumH.hasMoreElements() && ((Hashtable<Habilidad, Integer>) enumH).containsKey(habilidadExistente) ) // mientras haya elemento 
+		{
+			habilidadExistente=enumH.nextElement(); //sigo
+		}
+		habilidades.remove(habilidadExistente); // sin importarme total me trae null si la encontro o no la limpio.
```

Comment:
Con solo esa linea, ya te borra el elemento que querias
ahi te agregure para que handlees el error de si no se borro nada, fijate si habilidad.getNombre() existe
```suggestion
		Habilidad habilidadBorrada = habilidades.remove(habilidad); // sin importarme total me trae null si la encontro o no la limpio.
		if (habilidadBorrada == null) {
		    Logger.logError("El empleado " + this.nombre + " no tiene la habilidad " + habilidad.getNombre())
		}
```

Reply:
creo que lo entendi , e implemente para handlear el error